dbms/src/Flash/Planner/plans/PhysicalWindow.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Logger.h>
+#include <DataStreams/WindowBlockInputStream.h>
+#include <Flash/Coprocessor/DAGContext.h>
+#include <Flash/Coprocessor/DAGExpressionAnalyzer.h>
+#include <Flash/Coprocessor/DAGPipeline.h>
+#include <Flash/Coprocessor/InterpreterUtils.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Flash/Planner/PhysicalPlanHelper.h>
+#include <Flash/Planner/plans/PhysicalWindow.h>
+#include <Interpreters/Context.h>
+
+namespace DB
+{
+PhysicalPlanNodePtr PhysicalWindow::build(
+    const Context & context,
+    const String & executor_id,
+    const LoggerPtr & log,
+    const tipb::Window & window,
+    const FineGrainedShuffle & fine_grained_shuffle,
+    const PhysicalPlanNodePtr & child)
+{
+    assert(child);
+    child->disableRestoreConcurrency();
```

Comment:
child is `PhysicalSource <-- PhysicalFinalProjection <-- PhysicalWindowSort` and not `PhysicalWindowSort`.
So we need call `disableRestoreConcurrency` in `PhysicalWindowSort` and `PhysicalFinalProjection `.
After DAGQueryBlock removed, the plan tree will like `PhysicalWindow <-- PhysicalSort`

Reply:
Why If child is PhysicalSource <-- PhysicalFinalProjection <-- PhysicalWindowSort and not PhysicalWindowSort, we need call disableRestoreConcurrency in PhysicalWindowSort and PhysicalFinalProjection.